FitGoal/FitGoal/UserProfileConfiguration/WalkthroughIcon.swift

```diff
@@ -22,32 +22,33 @@ struct WalkthroughIcon {
         }
     }
     
+    private var iconNamePrefix: String {
+        return name.lowercased()
+    }
+    
     var name: String {
-        return iconType.rawValue.capitalized
+        switch iconType {
+        case .fitnessLevel(let fitnessLevel):
+            return fitnessLevel.rawValue.capitalized
+        case .gender(let gender):
+            return gender.rawValue.capitalized
+        }
     }
     
     var image: UIImage {
         switch iconType {
-        case .athletic, .normal, .obese, .skinny:
+        case .fitnessLevel:
             let coreName = "BodyShape"
-            let bodyShapeImageName = iconType.prefix + coreName + iconNameSuffix
+            let bodyShapeImageName = iconNamePrefix + coreName + iconNameSuffix
             return UIImage(imageLiteralResourceName: bodyShapeImageName)
-        case .female, .male:
-            let genderImageName = iconType.prefix + iconNameSuffix
+        case .gender:
+            let genderImageName = iconNamePrefix + iconNameSuffix
             return UIImage(imageLiteralResourceName: genderImageName)
         }
     }
 }
 
-enum WalkThroughIconType: String {
-    case skinny
-    case normal
-    case obese
-    case athletic
-    case female
-    case male
-    
-    var prefix: String {
-        return self.rawValue
-    }
+enum WalkThroughIconType {
+    case gender(gender: Gender)
+    case fitnessLevel(fitnessLevel: FitnessLevel)
```

Comment:
Not a fan of this enum. Can you at least remove the label for the associated value?